### DISCUSSION TITLE
Is StreamFactory threadsafe?

### DISCUSSION THREAD

**Discussion by Manfred73** [beanio/beanio#150] [General]
I'm using a StreamFactory and create a new instance in my method. I was wondering though if I can just inject a StreamFactory at class level and whether it is threadsafe.
I found some api documentation where it says DefaultStreamFactory is threadsafe, but I think that's for beanio.org (version 2.x): http://beanio.org/docs/api/org/beanio/parser/DefaultStreamFactory.html.
When I click the api link on the v3 documentation site I get a Page not found: https://beanio.github.io/api/index.html.

```
public class StreamFactoryProducer {
   @Produces      // necessary for Quarkus to inject with CDI or constructor injection
   @ApplicationScoped
   public StreamFactory produceStreamFactory() {
      return StreamFactory.newInstance();
   }
}

@ApplicationScoped
@AllArgsConstructor
public class FileProcessor {
   private final StreamFactory streamFactory;

   public Integer processRecordsInFile(BufferedReader br, String filename) {
      final var builder = new StreamBuilder("fileInput")
            .format("fixedlength")
            .ignoreUnidentifiedRecords()
            .addRecord(FileRecord.class);
      streamFactory.define(builder);             // Is this threadsafe???
      return processFile(br, filename);
}
```

**Comment by bjansen** [comment]
Hi,

Nothing has changed in v3 regarding `DefaultStreamFactory`, so I guess it's still thread-safe.

> When I click the api link on the v3 documentation site I get a Page not found: https://beanio.github.io/api/index.html.

Huh yeah, I should fix that (the joys of single page apps intercepting clicks).
